src/main/java/com/hubspot/imap/client/ImapClient.java

```diff
@@ -210,6 +212,14 @@ private void startKeepAlive() {
     return send(new ListCommand(context, query));
   }
 
+  public Future<ListResponse> extendedListWithReturnOption(String context, String query, ReturnOption returnOption){
```

Comment:
I am actually not a fan of adding super specific helpers like this to the IMAP client API, I'd prefer just having people construct the `ListCommand` themselves. This class is getting too polluted (and the line to construct the command is actually shorter).